src/WTG.Analyzers/Analyzers/LinqEnumerable/LinqEnumerableAnalyzer.cs

```diff
@@ -52,189 +55,205 @@ public static void CompilationStart(CompilationStartAnalysisContext context)
 				return;
 			}
 
+			/* exclusively checking the existence of the Enumerable.Append method because 
+			   Enumerable.Append and Enumerable.Prepend were released at the same time */
 			var hasEnumerableAppendMethod = HasEnumerableAppendMethod(context.Compilation);
 			var cache = new FileDetailCache();
 
 			context.RegisterSyntaxNodeAction(c => Analyze(c, cache, hasEnumerableAppendMethod), SyntaxKind.SimpleMemberAccessExpression);
 		}
 
-		static bool ContainsSingleElement (ExpressionSyntax? e)
-		{
-			switch (e?.Kind())
-			{
-				case SyntaxKind.ImplicitArrayCreationExpression:
-					return ((ImplicitArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ObjectCreationExpression:
-					return ((ObjectCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ArrayCreationExpression:
-					return ((ArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				default:
-					return false;
-			}
-		}
+		static bool ContainsSingleElement(ExpressionSyntax? e) => LinqEnumerableUtils.GetInitializer(e)?.Expressions.Count == 1;
 
 		public static bool LooksLikeAppend(InvocationExpressionSyntax invocation) => (invocation.ArgumentList.Arguments.Count == 1 ? ContainsSingleElement(invocation.ArgumentList.Arguments[0].Expression) : ContainsSingleElement(invocation.ArgumentList.Arguments[1].Expression));
 
 		public static bool LooksLikePrepend(InvocationExpressionSyntax invocation)
 		{
-			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
+			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression.TryGetExpressionFromParenthesizedExpression();
 
-			if (expression.IsKind(SyntaxKind.ParenthesizedExpression))
+			if (expression == null)
```

Comment:
Since we always fall back to the original expression after unwrapping parentheses, it would be better to have a method that already returns the fallback so we just call it and if there are no parentheses we get our original input expression back.

i.e.:

```csharp
var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression.UnwrapIfParenthesized());
// use expression without null-checks
```